pkg/agents/team3/emotions.go

```diff
@@ -81,74 +50,97 @@ func changeInMood(a *CustomAgent3, pointsMin, pointsMax, direction int) {
 }
 
 // Function gets as input the mini and max change we want in, direction marks if we want it to go up or down
-func changeInMorality(a *CustomAgent3, pointsMin, pointsMax, direction int) {
+func (a *CustomAgent3) changeInMorality(pointsMin, pointsMax, direction int) {
 	points := rand.Intn(pointsMax-pointsMin) + pointsMin
 	if direction < 0 {
 		a.vars.morality -= points
 		if a.vars.morality < 0 {
 			a.vars.morality = 0
 		}
-	}
-	if direction > 0 {
+	} else {
 		a.vars.morality += points
 		if a.vars.morality > 100 {
 			a.vars.morality = 100
 		}
 	}
 }
 
+func (a *CustomAgent3) changeInStubbornness(change, direction int) {
+	if direction < 0 {
+		a.vars.stubbornness -= change
+		if a.vars.stubbornness < 0 {
+			a.vars.stubbornness = 0
+		}
+	} else {
+		a.vars.stubbornness += change
+		if a.vars.stubbornness > 75 {
+			a.vars.stubbornness = 75
+		}
+	}
+}
+
 // Updates mood and morality when the floor is changed. Called at the start of each day.
-//TO DO: Trial this and do sth different.
-func changeNewDay(a *CustomAgent3) {
+func (a *CustomAgent3) changeNewDay() {
 	a.knowledge.lastHP = a.HP()
-	if a.HP() < 50 {
-		changeInMorality(a, 10, 15, -1)
-		changeInMood(a, 10, 15, -1)
+	a.knowledge.agentAge = a.BaseAgent().Age()
+	if a.HP() < 25 {
+		a.changeInMorality(5, 10, -1)
+		a.changeInMood(8, 16, -1)
+	} else if a.HP() < 40 {
+		a.changeInMorality(0, 4, -1)
+		a.changeInMood(1, 6, -1)
+	} else if a.HP() < 75 {
+		a.changeInMorality(0, 4, 1)
+		a.changeInMood(1, 6, 1)
+	} else {
+		a.changeInMorality(3, 7, 1)
+		a.changeInMood(5, 10, 1)
+	}
+	a.knowledge.foodMovingAvg = float64(a.knowledge.foodMovingAvg)*0.9 + float64(a.knowledge.foodLastEaten)*0.1
+}
+
+func (a *CustomAgent3) reshuffleEstimator() {
+	if a.knowledge.reshuffleEst == -1 {
+		a.knowledge.reshuffleEst = a.Age()
 	} else {
-		changeInMorality(a, 10, 15, 1)
-		changeInMood(a, 10, 15, 1)
+		if a.Age()-a.knowledge.reshuffleEst < a.knowledge.reshuffleEst { //attempt to correct in case we resshufle to same floor twice
+			a.knowledge.reshuffleEst = a.Age() - a.knowledge.reshuffleEst
+		}
 	}
```

Comment:
MINOR: maybe put this in an `else if`? (lol idt it would matter very much though)